hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java

```diff
@@ -247,6 +248,8 @@ public HazelcastClientInstanceImpl(ClientConfig config,
         long backofftimeoutMs = properties.getLong(BACKPRESSURE_BACKOFF_TIMEOUT_MILLIS);
         if (maxAllowedConcurrentInvocations == Integer.MAX_VALUE) {
             callIdSequence = new CallIdSequenceWithoutBackpressure();
+        } else if (backofftimeoutMs <= 0) {
```

Comment:
The client and server should share same logic to create the appropriate CallIdSequence; so perhaps pull out a factory method that can be shared between client/server.

Reply:
OK, I introduced the CallIdFactory which is called from both server and client side.